its/ruling/src/test/resources/expected/kotlin-S1144.json

```diff
@@ -0,0 +1,82 @@
+{
+'kotlin-project:corda/core/src/main/kotlin/net/corda/core/internal/ResolveTransactionsFlow.kt':[
+42,
```

Comment:
That looks a lot like a FP. A safer approach would be to work at file level rather than at class level: for each private method in a file, check whether its name is used somewhere else in the same file.

Reply:
Actually, here the problem comes from the fact that there are missing elements in this AST.
This stems from: `class A(a: int) { constructor(): this(getValue(0)) {} }` -> the delegation of the secondary constructor to the primary constructor (`: this(getValue(0))`) is missing from AST, so this FP arises from the fact that the use of the private function is inside the delegation call.
It is the same for delegation to superclass constructor when there is no primary constructor `class B: A { constructor(a: Int): super(getValue(a)) }`

Reply:
This FP will be fixed when fixing the AST creation (https://jira.sonarsource.com/browse/SONARSLANG-70)